### ISSUE TITLE
indent & jsx: no indent validation in some scenarios

### ISSUE THREAD

**Issue by lukeapage** [eslint/eslint#8912]
**Tell us about your environment**

* **ESLint Version:** 4.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default - reproduced on the web demo

**What did you do? Please include the actual source code causing the issue.**

```js
function a(b) {
    return (<div>
        {
            (b &&
        <C
            size="normal"
        />
            )
        }
    </div>);
}
```

**What did you expect to happen?**

There is no error on the indentation of <C - I can put it at any indentation I like without error.

Also, not sure if its that the web demo hasn't been upgraded or I just can't reproduce there, but with brackets, I have to do this (parenthesis and brackets aligned):

```js
function a(b) {
    return (<div>
        {
        (b &&
            <C
                size="normal"
            />
        )
        }
    </div>);
}
```

and without the brackets it is ok

```js
function a(b) {
    return (<div>
        {
            b &&
                <C
                    size="normal"
                />
        }
    </div>);
}
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, this is a duplicate of https://github.com/eslint/eslint/issues/8443.
